colibre/registration.py

```diff
@@ -118,3 +118,77 @@
+            dust_fields.append(getattr(catalogue.dust_mass_fractions, sub_path))
+    total_dust_fraction = sum(dust_fields)
+except AttributeError:
+    total_dust_fraction = np.zeros(stellar_mass.size)
+
+total_dust_mass = total_dust_fraction * catalogue.masses.m_star
+total_dust_mass.name = "$M_{\\rm dust}$ not found"
+
+
+setattr(self, f"total_dust_masses", total_dust_mass)
+
+# species fraction properties
+gas_mass = catalogue.masses.m_gas
+gal_area = (
+    2 * np.pi * catalogue.projected_apertures.projected_1_rhalfmass_star_100_kpc ** 2
+)
+mstar_100 = catalogue.projected_apertures.projected_1_mass_star_100_kpc
+
+self.xgass_galaxy_selection = np.logical_and(
+    catalogue.apertures.mass_star_100_kpc > unyt.unyt_quantity(10 ** 9, "Solar_Mass"),
+    catalogue.apertures.mass_star_100_kpc
+    < unyt.unyt_quantity(10 ** (11.5), "Solar_Mass"),
+)
+
+self.xcoldgass_galaxy_selection = np.logical_and(
+    catalogue.apertures.mass_star_100_kpc > unyt.unyt_quantity(10 ** 9, "Solar_Mass"),
+    catalogue.apertures.mass_star_100_kpc
+    < unyt.unyt_quantity(10 ** (11.5), "Solar_Mass"),
+)
+
+self.mu_star = mstar_100 / gal_area
+self.mu_star.name = "$\\pi R_{*, 100 {\\rm kpc}}^2 / M_{*, 100 {\\rm kpc}}$"
+
+self.neutral_hydrogen_mass = catalogue.masses.m_star * 0.0
+self.molecular_hydrogen_mass = catalogue.masses.m_star * 0.0
+self.h2_to_stellar_mass = catalogue.apertures.zmet_star_100_kpc * 0.0
+self.hi_to_stellar_mass = catalogue.apertures.zmet_star_100_kpc * 0.0
+
+self.neutral_hydrogen_mass.name = "HI Mass (100 kpc)"
+self.hi_to_stellar_mass.name = "HI to Stellar Mass Fraction (100 kpc)"
+self.molecular_hydrogen_mass.name = "H$_2$ Mass (100 kpc)"
+self.h2_to_stellar_mass.name = "H$_2$ to Stellar Mass Fraction (100 kpc)"
+
+try:
+    H_frac = catalogue.element_mass_fractions.element_0
+    # NB: this is designed for CHIMES species arrays
+    try:
+        self.neutral_hydrogen_mass = (
+            gas_mass * H_frac * catalogue.species_fractions.species_1
+        )
+        self.hi_to_stellar_mass = self.neutral_hydrogen_mass / catalogue.masses.m_star
+    except AttributeError:
+        self.neutral_hydrogen_mass.name += " not found (no species field)"
+        self.hi_to_stellar_mass.name += " not calculable (no species field)"
+    try:
+        self.molecular_hydrogen_mass = (
+            gas_mass * H_frac * catalogue.species_fractions.species_7
+        )
+        self.h2_to_stellar_mass = self.molecular_hydrogen_mass / catalogue.masses.m_star
+    except AttributeError:
+        self.molecular_hydrogen_mass.name += " not found (no species field)"
+        self.h2_to_stellar_mass.name += " not calculable (no species field)"
+
+except AttributeError:
+    self.molecular_hydrogen_mass.name += " not found (no H abundance)"
+    self.neutral_hydrogen_mass.name += " not found (no H abundance)"
+    self.hi_to_stellar_mass.name += " not calculable (no H abundance)"
+    self.h2_to_stellar_mass.name += " not calculable (no H abundance)"
```

Comment:
There are a number of things here:

H2 to stellar mass etc. doesn't use the 100 kpc aperture stellar mass (that is apertures.mass_star_100_kpc). Please also include aperture size in these names.

Alongside this, the code path is a bit messy - I don't think we need things to be nested and I like to avoid string concatenation with +s wherever possible.

```
try:
   H_frac = catalogue.element_mass_fractions.element_0
   hydrogen_frac_error = ""
except:
   H_frac = 0.0
   hydrogen_frac_error = "(no H abundance)"
   
   
# Test for CHIMES arrays
try:
   species_1 = catalogue.species_fractions.species_1
   species_7 = catalogue.species_fractions.species_7
   species_frac_error = ""
except:
   species_1 = 0.0
   species_7 = 0.0
   species_frac_error = "(no species field)"
   
total_error = f" {species_frac_error} {hydrogen_frac_error}"

self.neutral_hydrogen_mass = gas_mass * H_frac * species_1
self.hi_mass_to_stellar_mass_100_kpc = (
   self.neutral_hydrogen_mass / catalogue.apertures.mass_star_100_kpc
)
self.molecular_hydrogen_mass = gas_mass * H_frac * species_7
self.h2_mass_to_stellar_mass_100_kpc = (
   self.molecular_hydrogen_mass / catalogue.apertures.mass_star_100_kpc
)

self.neutral_hydrogen_mass.name = f"HI Mass (100 kpc) {total_error}"
self.hi_to_stellar_mass.name = f"HI to Stellar Mass Fraction (100 kpc) {total_error}"
self.molecular_hydrogen_mass.name = f"H$_2$ Mass (100 kpc) {total_error}"
self.h2_to_stellar_mass.name = f"H$_2$ to Stellar Mass Fraction (100 kpc) {total_error}"
```